src/App.tsx

```diff
@@ -1,24 +1,176 @@
+import { deleteTodo, getTodos } from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { ErrorList } from './components/ErrorList/ErrorList';
+import {
+  TodoFilter,
+  FilterTodoStatus,
+} from './components/TodoFilter/TodoFilter';
+import { TodoForm, TodoFormProps } from './components/TodoForm/TodoForm';
 
-const USER_ID = 0;
+import { ErrorTypes } from './types/ErrorTypes';
+
+const USER_ID = 6848;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+
+  const [filter, setFilter] = useState<FilterTodoStatus>(FilterTodoStatus.ALL);
+  const [errors, setErrors] = useState<string>('');
+  const [deletedTodosIds, setDeletedTodosIds] = useState<number[]>([]);
+
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((result) => setTodos(result))
+      .catch(() => setErrors(ErrorTypes.Load));
+  }, []);
+
+  const activeTodos = useMemo(
+    () => todos.filter((todo) => !todo.completed),
+    [todos],
+  );
+
+  const hasCompletedTodos = useMemo(
+    () => todos.some((todo) => todo.completed),
+    [todos],
+  );
+
+  const visibleTodos = useMemo(() => {
+    return filter === FilterTodoStatus.ALL
+      ? todos
+      : todos.filter((todo) => {
+        switch (filter) {
+          case FilterTodoStatus.ACTIVE:
+            return !todo.completed;
+
+          case FilterTodoStatus.COMPLETED:
+            return todo.completed;
+
+          default:
+            return true;
+        }
+      });
+  }, [todos, filter]);
+
+  const handleSuccess = useCallback<TodoFormProps['onSuccess']>((todo) => {
+    setTodos((prevTodos) => [...prevTodos, todo]);
+    setTempTodo(null);
+  }, []);
+
+  const handleErrorAdd = useCallback<TodoFormProps['onError']>(
+    (errorMessage: string) => {
+      setTempTodo(null);
+
+      setErrors(errorMessage);
+    },
+    [],
+  );
+
+  const handleTempTodo = useCallback<TodoFormProps['onTempTodo']>((todo) => {
+    setTempTodo(todo);
+  }, []);
+
+  const handleDeleteTodo = async (todoId: number) => {
+    try {
+      setDeletedTodosIds((prevState) => [...prevState, todoId]);
+      const response = await deleteTodo(todoId);
+
+      if (response === 0) {
+        throw new Error(ErrorTypes.Delete);
+      }
+
+      const newTodos = todos.filter((todo) => todo.id !== todoId);
+
+      setTodos([...newTodos]);
+    } catch {
+      setErrors(ErrorTypes.Delete);
+    } finally {
+      setDeletedTodosIds(
+        deletedTodosIds.filter((deletedTodoId) => deletedTodoId !== todoId),
+      );
+    }
+  };
+
+  const resetErrorHandler = useCallback(() => {
+    setErrors('');
+  }, []);
+
+  const handleClearCompleted = async () => {
+    const tasks: Promise<void>[] = [];
+
+    todos
+      .filter((todo) => todo.completed)
+      .forEach((todo) => {
+        handleDeleteTodo(todo.id);
+        tasks.push(deleteTodo(todo.id) as Promise<void>);
```

Comment:
try to avoid as in typeScript
```suggestion
        tasks.push(deleteTodo(todo.id));
```